mempool/clist_mempool.go

```diff
@@ -539,6 +539,20 @@ func (mem *CListMempool) ReapMaxTxs(max int) types.Txs {
 	return txs
 }
 
+// FilterTx returns the types.Tx with the given hash if found in the mempool, otherwise returns nil.
+func (mem *CListMempool) FilterTx(hash []byte) types.Tx {
+	mem.updateMtx.RLock()
+	defer mem.updateMtx.RUnlock()
+
+	for e := mem.txs.Front(); e != nil; e = e.Next() {
```

Comment:
There is no need to traverse the whole clist to find the transaction. (This is locking the mempool and it could be a DoS attack vector). The function `InMempool` should be used instead.